mne/annotations.py

```diff
@@ -877,7 +907,30 @@ def _check_event_id(event_id, raw):
     elif callable(event_id) or isinstance(event_id, dict):
         return event_id
     else:
-        raise ValueError('Invalid input event_id')
+        raise ValueError('Invalid type for event_id (should be None, str, '
+                         'dict or callable). Got {}'.format(type(event_id)))
+
+
+def _check_event_description(event_desc, events):
+    """Check event_id and convert to default format."""
+    if event_desc is None:  # convert to int to make typing-checks happy
+        event_desc = list(np.unique(events[:, 2]))
+
+    if isinstance(event_desc, dict):
+        for val in event_desc.values():
+            _validate_type(val, (str, None), 'Event names')
+    elif isinstance(event_desc, collections.Iterable):
+        event_desc = np.asarray(event_desc)
+        assert event_desc.ndim == 1
```

Comment:
It's not a good idea to use `assert` for statements that you expect users might hit, as they can be ignored with `python -O`. Better to do:
```
if event_desc.ndim != 1:
    raise ValueError('event_desc must be 1D, got shape %s' % (event_desc.shape,))
```

It's fine to use `assert` for code in tests (below), though, since pytest does nice things with it. We also sometimes use `assert` statements for code dev to assert things that should be true / users should never hit, kind of like "active comments".